app/src/main/java/net/squanchy/proximity/preconditions/TaskLauncherFactory.java

```diff
@@ -0,0 +1,84 @@
+package net.squanchy.proximity.preconditions;
+
+import android.app.Activity;
+import android.app.Fragment;
+import android.content.Intent;
+import android.content.IntentSender;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.AppCompatActivity;
+
+import timber.log.Timber;
+
+public final class TaskLauncherFactory {
+
+    private TaskLauncherFactory() {
+        // Not instantiable
+    }
+
+    public static TaskLauncher forFragment(Fragment fragment, Activity activity) {
+        return new TaskLauncher() {
+            @Override
+            public void startActivityForResult(Intent intent, int requestCode) {
+                fragment.startActivityForResult(intent, requestCode);
+            }
+
+            @Override
+            public boolean permissionGranted(String permission) {
+                return ActivityCompat.checkSelfPermission(activity, permission) == PackageManager.PERMISSION_GRANTED;
+            }
+
+            @Override
+            public void requestPermissions(String[] permissions, int requestCode) {
+                if (isMarshmallowOrLater()) {
+                    fragment.requestPermissions(permissions, requestCode);
+                } else {
+                    Timber.e(new IllegalStateException("Trying to request runtime permission on Android pre-M"));
```

Comment:
I'm not sure I follow? I don't want _anything_ that is `Timber.e` to get converted into an exception and go to Crashlytics.